packages/manager/apps/dedicated/client/app/billing/common/messageParser.js

```diff
@@ -1,85 +1,83 @@
-      PARTIAL: ALERT_LEVELS.WARNING,
-      WARNING: ALERT_LEVELS.WARNING,
-    };
+  const ALERT_MAP = {
+    ERROR: ALERT_LEVELS.ERROR,
+    INFO: ALERT_LEVELS.SUCCESS,
+    OK: ALERT_LEVELS.SUCCESS,
+    PARTIAL: ALERT_LEVELS.WARNING,
+    WARNING: ALERT_LEVELS.WARNING,
+  };
 
-    /**
-     * @param {string} message - text
-     * @param {object|string} data - additional data to parse
-     * @return {object}
-     */
-    return function billingMessageParser(message, data) {
-      let alertType = '';
-      if (!data) {
-        return { message, alertType: '' };
-      }
+  /**
+   * @param {string} message - text
+   * @param {object|string} data - additional data to parse
+   * @return {object}
+   */
+  return function billingMessageParser(message, data) {
+    let alertType = '';
+    if (!data) {
+      return { message, alertType: '' };
+    }
 
-      if (data === 'true') {
-        return { message, alertType: ALERT_LEVELS.SUCCESS };
-      }
+    if (data === 'true') {
+      return { message, alertType: ALERT_LEVELS.SUCCESS };
+    }
 
-      if (!isEmpty(message)) {
-        return {
-          message,
-          alertType: get(ALERT_MAP, data.type, ''),
-        };
-      }
+    if (!isEmpty(message)) {
+      return {
+        message,
+        alertType: get(ALERT_MAP, data.type, ''),
+      };
+    }
 
-      if (data.message) {
-        return {
-          message: `${message}(${data.message})`,
-          alertType: ALERT_MAP[data.type] || '',
-        };
-      }
+    if (data.message) {
+      return {
+        message: `${message}(${data.message})`,
+        alertType: ALERT_MAP[data.type] || '',
+      };
+    }
 
-      if (data.messages && data.messages.length > 0) {
-        const messageDetails = data.messages
-          .map((_message) => `${_message.id} : ${_message.message}`)
-          .join(', ');
+    if (data.messages && data.messages.length > 0) {
+      const messageDetails = data.messages
+        .map((_message) => `${_message.id} : ${_message.message}`)
+        .join(', ');
 
-        return {
-          message: `${message} (${messageDetails})`,
-          alertType: ALERT_MAP[data.state] || '',
-        };
-      }
+      return {
+        message: `${message} (${messageDetails})`,
+        alertType: ALERT_MAP[data.state] || '',
+      };
+    }
 
-      if (data.idTask && angular.isString(data.state)) {
-        const stateUppercased = data.state.toUpperCase();
-        switch (stateUppercased) {
-          case 'BLOCKED':
-          case 'CANCELLED':
-          case 'PAUSED':
-          case 'ERROR':
-            alertType = 'alert-danger';
-            break;
-          case 'WAITING_ACK':
-          case 'WAITINGACK':
-          case 'DOING':
-          default:
-            alertType = 'alert';
-            break;
-          case 'TODO':
-          case 'DONE':
-            alertType = 'alert-success';
-            break;
-        }
+    if (data.idTask && angular.isString(data.state)) {
```

Comment:
what is the difference with native `typeOf` ?

Reply:
i think there is none : https://github.com/angular/angular.js/blob/master/src/Angular.js#L568